app/models/in_progress_form.rb

```diff
@@ -1,8 +1,16 @@
 # frozen_string_literal: true
 class InProgressForm < ActiveRecord::Base
   attr_encrypted :form_data, key: Settings.db_encryption_key
+  validates(:form_data, presence: true)
+  before_save :serialize_form_data
 
   def self.form_for_user(form_id, user)
     InProgressForm.find_by(form_id: form_id, user_uuid: user.uuid)
   end
+
+  private
+
+  def serialize_form_data
+    self.form_data = form_data.to_json unless form_data.is_a?(String)
```

Comment:
if frontend can send us form_data already JSON stringified then there will be a minimum amount of conversions happening.

in that case we should send back the form_data as a JSON string inside a JSON string, if we parse it first then olive_branch will try to convert all the keys to camelcase which theoretically shouldn't do anything but it will slow down each request.